virttest/utils_test/libvirt.py

```diff
@@ -1059,23 +1059,39 @@ def pre_pool(self, pool_name, pool_type, pool_target, emulated_image,
                                                                nfs_path)
         elif pool_type == "iscsi":
             ip_protocal = kwargs.get('ip_protocal', "ipv4")
+            iscsi_chap_user = kwargs.get('iscsi_chap_user', None)
+            iscsi_chap_password = kwargs.get('iscsi_chap_password', None)
+            iscsi_secret_usage = kwargs.get('iscsi_secret_usage', None)
             if ip_protocal == "ipv6":
                 ip_addr = "::1"
             else:
                 ip_addr = "127.0.0.1"
-            setup_or_cleanup_iscsi(is_setup=True,
-                                   emulated_image=emulated_image,
-                                   image_size=image_size,
-                                   portal_ip=ip_addr)
+            if iscsi_chap_user and iscsi_chap_password and iscsi_secret_usage:
+                logging.debug("setup iscsi pool with chap authentication")
+                setup_or_cleanup_iscsi(is_setup=True,
```

Comment:
when prepare a valid iscsi pool with auth enabled, we'll need
-  chapuser/password for iscsi setup
- a libvirt secret for pool preparation
So the logic here is when missing anyone of chapuser/password/secret_usage, we'll prepare a unauthenticated iscsi pool, and debug info will show that "pool prepared without authentication (due to not fully provide chap user/password and libvirt secret usage)"
If we put the setup_or_cleanup_iscsi out of the condition, and user misconfigs the kwargs, such as only provide chap user and password, but not a libvirt secret, the iscsi server will still be authenticated but the pool cannot be started due to no libvirt secret provided. 
So my thought is to avoid pre_pool failures here, but provide detailed debug info to let user know if the authentication enabled or not, and why.